app/Models/Foundation/Main/Strategies/MemberSummitStrategyFactory.php

```diff
@@ -0,0 +1,34 @@
+<?php namespace App\Models\Foundation\Main\Strategies;
+
+/**
+ * Copyright 2024 OpenStack Foundation
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+use App\Models\Foundation\Main\IGroup;
+use LaravelDoctrine\ORM\Facades\Registry;
+use models\main\Member;
+
+class MemberSummitStrategyFactory
+{
+    /**
+     * @param Member $member
+     * @param String $entity_manager
+     * @return IMemberSummitStrategy
+     */
+    public static function getMemberSummitStrategy(Member $member, String $entity_manager): IMemberSummitStrategy
+    {
+        if ($member->belongsToGroup(IGroup::Sponsors)) {
+            return new SponsorMemberSummitStrategy($member->getId(), Registry::getManager($entity_manager));
```

Comment:
dont do , Registry::getManager($entity_manager)
just use
 Registry::getManager(SilverstripeBaseModel::EntityManager) inside each strategy avoid unnecessary params on the call stack